### PR TITLE
added cmd support

### PR DESCRIPTION
**Pull request by babajikomali**

### CODE REVIEW

File: shelLF/cli.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package shelLF
```
Comment (Durga-chikkala) [inline shelLF/cli.go]: change the dir name to cmd

File: shelLF/cli.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+import (
+	"flag"
+	"log"
+	"os"
+	"time"
+)
+
+func NewCLI(toolName string) *CLI {
```
Comment (Durga-chikkala) [inline shelLF/cli.go]: no need to take toolName  and func name as NewCMD()

File: shelLF/cli.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	return &CLI{toolName: toolName, commands: commandMap}
+}
+
+func (cli *CLI) AddCommand(cmd Command) {
+	cli.commands[cmd.Name] = &cmd
+}
+
+func (cli *CLI) printUsage() {
+	log.Printf("Usage: %s <command> [options]", cli.toolName)
+	log.Println("Available commands:")
```
Comment (Durga-chikkala) [inline shelLF/cli.go]: use goLF loggers

File: shelLF/cli.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	for _, cmd := range cli.commands {
+		log.Printf("  %s\t%s\n", cmd.Name, cmd.Description)
```
Comment (Durga-chikkala) [inline shelLF/cli.go]: can we format in a better way :

what's my command : %v ,  --> Name 
what I do: %v. -> Description: %v,
what I accept: %v

File: shelLF/cli.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+func (cli *CLI) Run() {
+	if len(os.Args) <= 1 || os.Args[1] == "help" {
+		cli.printUsage()
+		os.Exit(1)
+	}
+
+	cmdName := os.Args[1]
+	cmd, ok := cli.commands[cmdName]
+	if ok {
+		if err := cmd.Flags.Parse(os.Args[2:]); err != nil {
```
Comment (Durga-chikkala) [inline shelLF/cli.go]: format the code properly

File: cmd/cli.go [hunk 6]
```diff
@@ -1,42 +1,62 @@
...
 func (cli *CLI) AddCommand(cmd Command) {
 	cli.commands[cmd.Name] = &cmd
 }
 
 func (cli *CLI) printUsage() {
-	log.Printf("Usage: %s <command> [options]", cli.toolName)
-	log.Println("Available commands:")
+	logger := logger.NewCustomLogger()
+	logger.Infof("Usage: %s <command> [options]", cli.toolName)
+	logger.Infof("Available commands:")
```
Comment (Durga-chikkala) [inline cmd/cli.go]: use fmt.Println or Printf here  because logs will give time as well which is not needed

File: cmd/cli.go [hunk 7]
```diff
@@ -1,42 +1,62 @@
...
 	for _, cmd := range cli.commands {
-		log.Printf("  %s\t%s\n", cmd.Name, cmd.Description)
+		logger.Infof("What is my command: %s\n", cmd.Name)
+		logger.Infof("What I do: %s\n", cmd.Description)
+		logger.Infof("What I accept:")
+		cmd.Flags.PrintDefaults()
```
Comment (Durga-chikkala) [inline cmd/cli.go]: same here and value is missing for 3rd one

**Review by DattaSaiTejaTadi — APPROVED** [review]

**Review by Durga-chikkala — APPROVED** [review]

**Pull request merged**